common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -554,6 +568,49 @@ def should_force_account_creation():
         # We know they did not just login, because the login process rejects unverified users.
 
 
+def create_user(strategy, details, user=None, autoprovision=False, *args, **kwargs):
+    """
+    Creates user passing extra fields if in autoprovisioning mode
+    """
+    if user:
+        return {'is_new': False}
+
+    fields = dict(
+        (name, kwargs.get(name) or details.get(name))
+        for name in (strategy.setting('USER_FIELDS', USER_FIELDS))
+    )
+
+    if not fields:
+        return
+
+    if not autoprovision:
+        return {
+            'is_new': True,
+            'user': strategy.create_user(**fields)
+        }
+
+    extra_fields = dict(
+        (name, kwargs.get(name) or details.get(name, None))
+        for name in (strategy.setting('EXTRA_USER_FIELDS', []))
+    )
+
+    fields.update(extra_fields)
+    fields['autoprovision'] = True
+
+    try:
+        # Mocking partial pipeline request so that create_account_with_params used in strategy to create a user
+        # correctly get backend and provider settings
+        save_status_to_session(strategy, details=details, user=user, autoprovision=autoprovision, *args, **kwargs)
```

Comment:
@bradenmacdonald providing password in strategy was my initial atempt. Unfortunately, that's not the only way `create_account_with_params` uses pipeline: https://github.com/edx-solutions/edx-platform/blob/cypress-rebase/common/djangoapps/student/views.py#L1628-L1629. So the reason for this is to access skip_email_verification parameter of provider.